File: book/petrol-pump/pump/SButton.java
package pump;

import sodium.*;
import javax.swing.JButton;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class SButton extends JButton
{
    public SButton(String label)
    {
        super(label);
        EventSink<Unit> eClickedSink = new EventSink<>();
        this.eClicked = eClickedSink;
        addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                eClickedSink.send(Unit.UNIT);
            }
        });
    }

    public final Event<Unit> eClicked;
}

File: c#/Book/PetrolPump/PetrolPump/PetrolPump/Pump/SButton.cs
using System;
using System.Windows.Controls;

using Sodium;

namespace PetrolPump.Pump
{
  public class SButton : Button
  {
    public SButton(String label)
    {
      Content = label;
      EventSink<Unit> eClickedSink = new EventSink<Unit>();
      EClicked = eClickedSink;
      Click += (sender, args) => { eClickedSink.Send(Unit.UNIT); };
    }

    public readonly Event<Unit> EClicked;
  }
}
